### PR TITLE
#98 use the native time picker when on mobile devices

### PR DESCRIPTION
**Pull request by rezk2ll**
tested on android

event date edition / creation demo: https://streamable.com/6vl162
event date suggestion demo: https://streamable.com/ztgxh3

### CODE REVIEW

File: src/esn.calendar.libs/app/components/event-date-edition/event-date-edition.controller.js [hunk 1]
```diff
@@ -113,13 +121,22 @@ function calEventDateEditionController(esnI18nDateFormatService, esnDatetimeServ
...
+    _refreshTimeInputs();
   }
 
   function onEndDateTimeChange() {
+    if (self.isMobile) {
+      self.end.set({
+        hour: self.endTime.getHours(),
+        minute: self.endTime.getMinutes()
+      });
+    }
```
Comment (huy-ta) [inline src/esn.calendar.libs/app/components/event-date-edition/event-date-edition.controller.js:133]: Why this logic?

File: src/esn.calendar.libs/app/components/event-date-edition/event-date-edition.controller.js [hunk 2]
```diff
@@ -166,4 +183,9 @@ function calEventDateEditionController(esnI18nDateFormatService, esnDatetimeServ
   function _updateMinEndDate() {
     self.minEndDate = getMinEndDate();
   }
+
+  function _refreshTimeInputs() {
+    self.startTime = self.start.toDate();
+    self.endTime = self.end.toDate();
+  }
```
Comment (huy-ta) [inline src/esn.calendar.libs/app/components/event-date-edition/event-date-edition.controller.js:190]: Why this logic?

File: src/esn.calendar.libs/app/components/event-date-edition/event-date-edition.controller.js (hunk 1 above)
Comment (rezk2ll) [reply src/esn.calendar.libs/app/components/event-date-edition/event-date-edition.controller.js:133]: the `onEndDateTimeChange` handler can be called by both the time selector and the mobile picker.

if we are in the mobile picker we get the hours and minutes from the `Date` object ( not `moment` object ).

the native time picker uses only `Date` objects and due to the differences between moment and Date I have to only set the Hours and minutes without worrying about the Timezone.

File: src/esn.calendar.libs/app/components/event-date-edition/event-date-edition.controller.js (hunk 2 above)
Comment (rezk2ll) [reply src/esn.calendar.libs/app/components/event-date-edition/event-date-edition.controller.js:190]: check https://github.com/OpenPaaS-Suite/esn-frontend-calendar/pull/201/files#r502385815

**Review by asmaNode — DISMISSED** [review]

File: src/esn.calendar.libs/app/components/event-date-edition/event-date-edition.pug [hunk 3]
```diff
@@ -24,7 +24,16 @@
...
-                disabled="ctrl.disabled"
+                disabled="ctrl.disabled",
+                ng-if="!ctrl.isMobile"
+              )
+              input.input.form-control(
+                type="time",
+                ng-model="ctrl.startTime",
+                ng-change="ctrl.onStartDateTimeChange()",
+                ng-blur="ctrl.dateOnBlurFn()",
+                ng-disabled="ctrl.full24HoursDay || ctrl.disabled",
```
Comment (renaudboyer) [inline src/esn.calendar.libs/app/components/event-date-edition/event-date-edition.pug:35]: Why ng-disabled="ctrl.full24HoursDay" ?
Line should be hidden in this case

Comment (rezk2ll) [reply src/esn.calendar.libs/app/components/event-date-edition/event-date-edition.pug]: fixed thanks.

File: src/esn.calendar.libs/app/components/event-date-suggestion/event-date-suggestion.controller.js [hunk 4]
```diff
@@ -79,4 +83,41 @@ function calEventDateSuggestionController(esnI18nDateFormatService, calMoment, e
...
+    if (!self.event.start || !self.event.start.isValid() || !self.event.end || !self.event.end.isValid()) {
+      return;
+    }
+
+    self.startTime = self.event.start.toDate();
+    self.endTime = self.event.end.toDate();
+
+    // Set the hours to avoid the timzone issues when converting a moment object to Date.
+    self.startTime.setHours(self.event.start.hours());
+    self.endTime.setHours(self.event.end.hours());
```
Comment (huy-ta) [inline src/esn.calendar.libs/app/components/event-date-suggestion/event-date-suggestion.controller.js:121]: 1. First, I think this logic should only impact mobile devices. From what I read here, it also affects every other device.
2. If I recall correctly, we have some timezones that are only +/- 30 minutes away from each other, so only setting hours may not be enough.

Comment (rezk2ll) [reply src/esn.calendar.libs/app/components/event-date-suggestion/event-date-suggestion.controller.js]: I agree with you on `2`, it's fixed now, we set the minutes also.
for `1`, `_setTimeInputs` is only related to the input models `startTime` and `endTime` that are only used with the native `time` fields ( that only shows for mobile ).

Comment (huy-ta) [reply src/esn.calendar.libs/app/components/event-date-suggestion/event-date-suggestion.controller.js]: I see, how about `_initMobileTimeInputs` for the function's name?

Comment (rezk2ll) [reply src/esn.calendar.libs/app/components/event-date-suggestion/event-date-suggestion.controller.js]: done

**Review by huy-ta — APPROVED** [review]

**Review by asmaNode — APPROVED** [review]

**Pull request merged**